repository/roverClient.js

```diff
@@ -0,0 +1,27 @@
+import axios from 'axios';
+import https from 'https';
+import config from '../config/config.js';
+
+const nasaRoverUrl = config.nasaRoverUrl;
+
+const getRoverPhotos = async (apiKey) => {
+  try {
+    const httpsAgent = new https.Agent({
+      rejectUnauthorized: false,
+    });
+
+    const response = await axios.get(nasaRoverUrl, {
+        params: {
+          api_key: apiKey,
+          sol: config.sol
+        },
+      httpsAgent: httpsAgent,
+    });
+
+    return response.data;
+  } catch (error) {
```

Comment:
we will handle all errors in the error handler and try..catch should be added only on controller layer